golem-test-framework/src/components/component_service/mod.rs

```diff
@@ -85,20 +86,17 @@ pub enum PluginServiceClient {
 }
 
 #[async_trait]
-pub trait ComponentService {
-    fn client_protocol(&self) -> GolemClientProtocol;
+trait ComponentServiceInternal {
     fn component_client(&self) -> ComponentServiceClient;
     fn plugin_client(&self) -> PluginServiceClient;
+    fn plugin_wasm_files_service(&self) -> Arc<PluginWasmFilesService>;
+}
 
+#[allow(private_bounds)]
```

Comment:
This is just here because I didn't make ComponentServiceInternal public.
This is perfectly fine, but as the name of ComponentServiceInternal is not exported, nobody outside this crate can implement the trait. I think this is desirable behaviour, but if we want to allow outside implementations we just need to export the internal trait